### ISSUE TITLE
could not find version of module

### ISSUE THREAD

**Issue by alidcast** [VinAIResearch/Anti-DreamBooth#5]
I know there are difficulties installing python dependencies, so apologies if this is not relevant to the project setup.

But I've followed the steps, activating an environment with conda and running pip install, but there are several modules not found, such as `triton` and `pytorch`. For example: 

```
ERROR: Could not find a version that satisfies the requirement triton==2.0.0 (from versions: none)
ERROR: No matching distribution found for triton==2.0.0
```

I don't have enough experience with Python to know if this is specific to my environment. It's pointing to correct versions, `python 3.9.0` and `pip 23.0.1`, both specific to the conda environment.

Please let me know if there's anything else that may be done. ChatGTP was letting me know I could try installing deps from other sources, but seems weird that the sources you guys provided wouldn't work.

**Comment by Luvata** [comment]
Sorry, the correct package version we used is `2.0.0.dev20221031`
You can install it by running the following command 
```bash
pip install triton==2.0.0.dev20221031
```

Please let us know if you have any further questions or issues.

**Comment by thuanz123** [comment]
Hi, we think that triton is quite new so may be some newest version does not support on macos. You can try using the newest triton available on macos

**Comment by thuanz123** [comment]
Yeah, we have just found out that triton only support Linux, so you can remove the triton from the requirements and try to see if our code still works. We would love to know your feedback, since we don't have any MacOS to test

**Comment by Luvata** [comment]
I'm sorry for not asking thoroughly about your system before providing the previous response.
Our experiments were conducted using Linux with Nvidia A100. We used xformers' efficient attention for faster training and lower memory footprint on Nvidia GPU, and triton is a dependency for xformers.

One thing you can try is to run [train_dreambooth_alone.sh](https://github.com/VinAIResearch/Anti-DreamBooth/blob/main/scripts/train_dreambooth_alone.sh#LL8) without xformers by removing line 8 `enable_xformers_memory_efficient_attention` and let us know if it still works.

For comparison, it took around 10 minutes to run `train_dreambooth_alone.sh` using A100 with efficient attention enabled. However, given that finetuning Dreambooth (or training Anti-Dreambooth) is a computationally intensive task that requires significant computing resources, it may take much longer or even be unfeasible to run on less powerful devices. So we highly recommend running our code on a similar system, such as Linux with an Nvidia A6000 or A100, for optimal performance.
